pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGateFunctors.hpp

```diff
@@ -671,34 +722,52 @@ void applyCZ(Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
              const std::vector<std::size_t> &wires,
              [[maybe_unused]] const bool inverse = false,
              [[maybe_unused]] const std::vector<PrecisionT> &params = {}) {
-    applyNC2Functor(
-        ExecutionSpace{}, arr_, num_qubits, wires,
-        KOKKOS_LAMBDA(Kokkos::View<Kokkos::complex<PrecisionT> *> arr,
-                      const std::size_t i00, const std::size_t i01,
-                      const std::size_t i10, const std::size_t i11) {
-            [[maybe_unused]] const auto i00_ = i00;
-            [[maybe_unused]] const auto i01_ = i01;
-            [[maybe_unused]] const auto i10_ = i10;
-            arr(i11) *= -1;
-        });
+    auto core_function = KOKKOS_LAMBDA(
+        Kokkos::View<Kokkos::complex<PrecisionT> *> arr, const std::size_t i00,
+        const std::size_t i01, const std::size_t i10, const std::size_t i11) {
```

Comment:
Why not adding `[[maybe_unused]]` to the signatures here?

Reply:
Discussed above